File: benchmarks/overflowdb/src/main/java/io/shiftleft/overflowdb/testdomains/gratefuldead/WrittenBy.java
package io.shiftleft.overflowdb.testdomains.gratefuldead;

import io.shiftleft.overflowdb.EdgeFactory;
import io.shiftleft.overflowdb.EdgeLayoutInformation;
import io.shiftleft.overflowdb.NodeRef;
import io.shiftleft.overflowdb.OdbGraph;
import io.shiftleft.overflowdb.OdbEdge;

import java.io.Serializable;
import java.util.Arrays;
import java.util.HashSet;

public class WrittenBy extends OdbEdge implements Serializable {
  public static final String LABEL = "writtenBy";
  public static final HashSet<String> PROPERTY_KEYS = new HashSet<>(Arrays.asList());

  public WrittenBy(OdbGraph graph, NodeRef outVertex, NodeRef inVertex) {
    super(graph, LABEL, outVertex, inVertex, PROPERTY_KEYS);
  }

  public static final EdgeLayoutInformation layoutInformation = new EdgeLayoutInformation(LABEL, PROPERTY_KEYS);

  public static EdgeFactory<WrittenBy> factory = new EdgeFactory<WrittenBy>() {
    @Override
    public String forLabel() {
      return WrittenBy.LABEL;
    }

    @Override
    public WrittenBy createEdge(OdbGraph graph, NodeRef outVertex, NodeRef inVertex) {
      return new WrittenBy(graph, outVertex, inVertex);
    }
  };
}

File: traversals/src/test/scala/io/shiftleft/overflowdb/testdomains/gratefuldead/WrittenBy.scala
package io.shiftleft.overflowdb.testdomains.gratefuldead

import io.shiftleft.overflowdb._
import scala.jdk.CollectionConverters._

object WrittenBy {
  val Label = "writtenBy"
  object PropertyKeys {
    val all: Set[String] = Set.empty
    val allAsJava: java.util.Set[String] = all.asJava
  }
  val layoutInformation = new EdgeLayoutInformation(Label, PropertyKeys.allAsJava)
  var factory: EdgeFactory[WrittenBy] = new EdgeFactory[WrittenBy] {
    override def forLabel(): String = WrittenBy.Label

    override def createEdge(graph: OdbGraph, outNode: NodeRef[OdbNode], inNode: NodeRef[OdbNode]): WrittenBy =
      new WrittenBy(graph, outNode, inNode)
  }
}

class WrittenBy(graph: OdbGraph, outVertex: NodeRef[_ <: OdbNode], inVertex: NodeRef[_ <: OdbNode])
  extends OdbEdge(graph, WrittenBy.Label, outVertex, inVertex, WrittenBy.PropertyKeys.allAsJava)
